pkg/strategy/dca2/maker.go

```diff
@@ -0,0 +1,114 @@
+package dca2
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/c9s/bbgo/pkg/exchange/retry"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+func (s *Strategy) placeDCAOrders(ctx context.Context) error {
+	s.logger.Infof("[DCA] start placing maker orders")
+	price, err := getBestPriceUntilSuccess(ctx, s.Session.Exchange, s.Symbol, s.Short)
+	if err != nil {
+		return err
+	}
+
+	orders, err := s.generateDCAOrders(s.Short, s.Budget, price, s.PriceDeviation, s.MaxOrderNum)
+	if err != nil {
+		return err
+	}
+
+	createdOrders, err := s.OrderExecutor.SubmitOrders(ctx, orders...)
+	if err != nil {
+		return err
+	}
+
+	s.debugOrders(createdOrders)
+
+	return nil
+}
+
+func getBestPriceUntilSuccess(ctx context.Context, ex types.Exchange, symbol string, short bool) (fixedpoint.Value, error) {
+	ticker, err := retry.QueryTickerUntilSuccessful(ctx, ex, symbol)
+	if err != nil {
+		return fixedpoint.Zero, err
+	}
+
+	if short {
+		return ticker.Buy, nil
+	} else {
+		return ticker.Sell, nil
+	}
+}
+
+func (s *Strategy) generateDCAOrders(short bool, budget, price, priceDeviation fixedpoint.Value, maxOrderNum int64) ([]types.SubmitOrder, error) {
+	// TODO: not implement short part yet
+	factor := fixedpoint.One.Sub(priceDeviation)
+	if short {
+		factor = fixedpoint.One.Add(priceDeviation)
+	}
+
+	// calculate all valid prices
+	var prices []fixedpoint.Value
+	for i := 0; i < int(maxOrderNum); i++ {
+		if i > 0 {
+			price = price.Mul(factor)
+		}
+		price = s.Market.TruncatePrice(price)
+		if price.Compare(s.Market.MinPrice) < 0 {
+			break
+		}
+
+		prices = append(prices, price)
+	}
+
+	notional, orderNum := calculateDCAMakerOrderNotionalAndNum(s.Market, short, budget, prices)
```

Comment:
generally you can remove all the "DCA" word in this package because you're already in the dca pacakge